File: app/src/test/java/eu/napcode/gonoteit/ui/login/UserValidatorTest.java
package eu.napcode.gonoteit.ui.login;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class UserValidatorTest {

    private UserValidator userValidator;

    @Before
    public void init() {
        userValidator = new UserValidator();
    }

    @Test
    public void testWithNotEmptyLoginAndPassword() {
        Assert.assertEquals(true, userValidator.isUserValid("user", "pass"));
    }

    @Test
    public void testWithEmptyLogin() {
        Assert.assertEquals(false, userValidator.isUserValid("", "pass"));
    }

    @Test
    public void testWithEmptyPassword() {
        Assert.assertEquals(false, userValidator.isUserValid("login", ""));
    }

    @Test
    public void testWithEmptyLoginAndPassword(){
        Assert.assertEquals(false, userValidator.isUserValid("", ""));
    }

    @Test
    public void testWithNullLogin(){
        Assert.assertEquals(false, userValidator.isUserValid(null, "pass"));
    }

    @Test
    public void testWithNullPass() {
        Assert.assertEquals(false, userValidator.isUserValid("login", null));
    }

    @Test
    public void testWithNullLoginAndPassword() {
        Assert.assertEquals(false, userValidator.isUserValid(null, null));
    }

    @Test
    public void testLoginValidation() {
        Assert.assertEquals(false, userValidator.isLoginValid(null));
        Assert.assertEquals(false, userValidator.isLoginValid(""));
        Assert.assertEquals(true, userValidator.isLoginValid("login"));
    }

    @Test
    public void testPasswordValidation() {
        Assert.assertEquals(false, userValidator.isPasswordValid(null));
        Assert.assertEquals(false, userValidator.isPasswordValid(""));
        Assert.assertEquals(true, userValidator.isPasswordValid("pass"));
    }

    @Test
    public void testTokenValidation() {
        Assert.assertEquals(false, userValidator.isTokenValid(null));
        Assert.assertEquals(false, userValidator.isTokenValid(""));
        Assert.assertEquals(true, userValidator.isTokenValid("token"));
    }
}

File: app/src/test/java/eu/napcode/gonoteit/ui/login/UserValidatorTest.kt
package eu.napcode.gonoteit.ui.login

import eu.napcode.gonoteit.app.utils.isLoginValid
import eu.napcode.gonoteit.app.utils.isPasswordValid
import eu.napcode.gonoteit.app.utils.isTokenValid
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner

import eu.napcode.gonoteit.app.utils.isUserValid

@RunWith(MockitoJUnitRunner::class)
class UserValidationTest {

    @Test
    fun testWithNotEmptyLoginAndPassword() {
        Assert.assertEquals(true, isUserValid("user", "pass"))
    }

    @Test
    fun testWithEmptyLogin() {
        Assert.assertEquals(false, isUserValid("", "pass"))
    }

    @Test
    fun testWithEmptyPassword() {
        Assert.assertEquals(false, isUserValid("login", ""))
    }

    @Test
    fun testWithEmptyLoginAndPassword() {
        Assert.assertEquals(false, isUserValid("", ""))
    }

    @Test
    fun testWithNullLogin() {
        Assert.assertEquals(false, isUserValid(null, "pass"))
    }

    @Test
    fun testWithNullPass() {
        Assert.assertEquals(false, isUserValid("login", null))
    }

    @Test
    fun testWithNullLoginAndPassword() {
        Assert.assertEquals(false, isUserValid(null, null))
    }

    @Test
    fun testLoginValidation() {
        Assert.assertEquals(false, isLoginValid(null))
        Assert.assertEquals(false, isLoginValid(""))
        Assert.assertEquals(true, isLoginValid("login"))
    }

    @Test
    fun testPasswordValidation() {
        Assert.assertEquals(false, isPasswordValid(null))
        Assert.assertEquals(false, isPasswordValid(""))
        Assert.assertEquals(true, isPasswordValid("pass"))
    }

    @Test
    fun testTokenValidation() {
        Assert.assertEquals(false, isTokenValid(null))
        Assert.assertEquals(false, isTokenValid(""))
        Assert.assertEquals(true, isTokenValid("token"))
    }
}